python2/koans/about_classes.py

```diff
@@ -125,36 +129,36 @@ def get_self(self):
             return self
 
         def __str__(self):
-            #
-            # Implement this!
-            #
-            return __
+            name = str(self._name)
+            return name
 
         def __repr__(self):
             return "<Dog named '" + self._name + "'>"
 
     def test_inside_a_method_self_refers_to_the_containing_object(self):
         fido = self.Dog6("Fido")
 
-        self.assertEqual(__, fido.get_self())  # Not a string!
+        self.assertEqual(fido, fido.get_self())  # Not a string!
```

Comment:
Is it clear why `fido` and `fido.get_self()` are equal?

Will the following pass?

```python
fido1 = self.Dog6("Fido")
fido2 = self.Dog6("Fido")
self.assertEqual(fido1, fido2.get_self())
```

Reply:
`fido` is the object created and the method `get_self` returns `self`, which refers again to the object itself.

`self.assertEqual(fido1, fido2.get_self())` would output an error because the `fido2.get_self()" is reffering to the object `fido2`, which is different from the object `fido1`, even if they are come from the same class.

Reply:
That's right!